modules/geo-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -0,0 +1,89 @@
+import {GLTFMaterialParser} from '@luma.gl/experimental';
+import {Model, pbr} from '@luma.gl/core';
+import {SimpleMeshLayer} from '@deck.gl/mesh-layers';
+
+import vs from './mesh-layer-vertex.glsl';
+import fs from './mesh-layer-fragment.glsl';
+
+function validateGeometryAttributes(attributes) {
+  const hasColorAttribute = attributes.COLOR_0 || attributes.colors;
+  if (!hasColorAttribute) {
+    attributes.colors = {constant: true, value: new Float32Array([1, 1, 1])};
+  }
+}
+
+const defaultProps = {
+  // PBR material object. _lighting must be pbr for this to work
+  pbrMaterial: {type: 'object', value: null}
+};
+
+export default class _MeshLayer extends SimpleMeshLayer {
+  getShaders() {
+    const shaders = super.getShaders();
+    const modules = shaders.modules;
+    modules.push(pbr);
+    return {...shaders, vs, fs};
+  }
+
+  updateState({props, oldProps, changeFlags}) {
+    super.updateState({props, oldProps, changeFlags});
+    if (props.pbrMaterial !== oldProps.pbrMaterial) {
+      this.updatePbrMaterialUniforms(props.pbrMaterial);
+    }
+  }
+
+  draw({uniforms}) {
```

Comment:
Pass the original argument to the parent class's method, in case more than `uniforms` is needed:

```js
draw(opts) {
  ...
  super.draw(opts);
}
```

Reply:
Done